servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -649,7 +663,35 @@ private static void logDecoderException(final DecoderException e,
             }
         }
 
-        private static void logUnexpectedException(final Throwable t, NettyHttpServerConnection connection) {
+        private static void logHttp2Exception(final Http2Exception e, final NettyHttpServerConnection connection) {
+            final Http2ErrorCode errorCode = e.errorCode();
+            if (CANCEL.equals(errorCode)) {
+                LOGGER.debug(
+                        "{} HTTP/2 stream was cancelled by a remote peer, most likely due to timeout or lost interest",
+                        connection, e);
+            } else if (NO_ERROR.equals(errorCode)) {
+                LOGGER.debug("{} HTTP/2 stream was closed because underlying connection closed due to GO_AWAY",
+                        connection, e);
+            } else if (SETTINGS_TIMEOUT.equals(errorCode)) {
+                LOGGER.warn("{} HTTP/2 stream was closed because underlying connection did not receive SETTINGS " +
+                        "acknowledgement on time", connection, e);
+            } else if (PROTOCOL_ERROR.equals(errorCode) || INTERNAL_ERROR.equals(errorCode) ||
+                    FLOW_CONTROL_ERROR.equals(errorCode) || STREAM_CLOSED.equals(errorCode) ||
+                    FRAME_SIZE_ERROR.equals(errorCode) || COMPRESSION_ERROR.equals(errorCode) ||
+                    INADEQUATE_SECURITY.equals(errorCode) || HTTP_1_1_REQUIRED.equals(errorCode)) {
+                LOGGER.warn("{} HTTP/2 stream failed with an error indicating client misbehavior", connection, e);
+            } else {
+                // REFUSED_STREAM & ENHANCE_YOUR_CALM - we don't support ServerPush, should only happen on client side
+                // CONNECT_ERROR - expected to happen only on client side
+                // Any other non-standard error code - we don't know how to handle it
+                LOGGER.warn("{} HTTP/2 stream failed with an error unexpected for the server-side", connection, e);
+            }
+            if (connection.nettyChannel().isOpen()) {
+                ChannelCloseUtils.close(connection.nettyChannel(), e);
```

Comment:
I would have not expected that a method called `logHttp2Exception` also performs the side effect of closing the connection in addition to logging. Maybe consider either renaming the method or refactoring the code?

Reply:
Moved closure outside `log*` methods: https://github.com/apple/servicetalk/pull/3206/commits/2f4c7f5596a6db42807a529ac82d53baeec845b9